grader_systems/src/context.rs

```diff
@@ -91,6 +92,31 @@ impl Context {
             ));
         }
 
+        if is_program_in_path("valgrind") {
+            if let Some(conf) = config.valgrind.clone() {
+                test.add_stage(Valgrind::new(
+                    ShellExecutor,
+                    conf,
+                    config.run.executable.clone(),
+                    config.run.args.clone(),
+                    config.run.stdin.clone(),
+                    config.run.timeout().clone(),
```

Comment:
You could just have the run config be the arg, and clone that instead of having to break it down like this?

Reply:
Yeah that's the right choice, will make those changes